src/os/bluestore/BlueStore.cc

```diff
@@ -2585,7 +3313,147 @@ void BlueStore::ExtentMap::dup(BlueStore* b, TransContext* txc,
     newo->onode.size = dstoff + length;
   }
   newo->extent_map.dirty_range(dstoff, length);
+  //_dup_writing needs cache lock
+  bcs->lock.unlock();
 }
+
+void BlueStore::ExtentMap::dup_esb(BlueStore* b, TransContext* txc,
+  CollectionRef& c, OnodeRef& oldo, OnodeRef& newo, uint64_t& srcoff,
```

Comment:
IIUC oldo is redundant here as well. dup_esb is always called on oldo, isn't it? 
Well I can understand a desire to leave the same interface as the original dup() and keep this parameter.
But I'd suggest to just have an assertion this->onode == oldo->get() (with proper comment) and get rid of 'oldo->*' usage wherever possible.
Renaming oldo to something like my_onode_ref would be nice to have too.

Reply:
Removed / refactored out references.